python/feathub/processors/spark/dataframe_builder/over_window_utils.py

```diff
@@ -0,0 +1,207 @@
+    ) -> "OverWindowDescriptor":
+        filter_expr = (
+            to_spark_sql_expr(window_agg_transform.filter_expr)
+            if window_agg_transform.filter_expr is not None
+            else None
+        )
+        return OverWindowDescriptor(
+            window_agg_transform.window_size,
+            window_agg_transform.limit,
+            window_agg_transform.group_by_keys,
+            filter_expr,
+        )
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, self.__class__)
+            and self.window_size == other.window_size
+            and self.limit == other.limit
+            and self.group_by_keys == other.group_by_keys
+            and self.filter_expr == other.filter_expr
+        )
+
+    def __hash__(self) -> int:
+        return hash(
+            (self.window_size, self.limit, tuple(self.group_by_keys), self.filter_expr)
+        )
+
+
+def _get_over_window_agg_column_list(
+    window_descriptor: "OverWindowDescriptor",
+    agg_descriptors: List["AggregationFieldDescriptor"],
+    window_spec: WindowSpec,
+) -> Dict[str, Column]:
+    return {
+        descriptor.field_name: _get_over_window_agg_select_expr(
+            descriptor.expr,
+            window_descriptor,
+            descriptor.agg_func,
+            window_spec,
+        )
+        .cast(descriptor.field_data_type)
+        .alias(descriptor.field_name)
+        for descriptor in agg_descriptors
+    }
+
+
+def _get_over_window_agg_select_expr(
+    expr: str,
+    over_window_descriptor: "OverWindowDescriptor",
+    agg_func: AggFunc,
+    window_spec: WindowSpec,
+) -> Column:
+
+    if (
+        over_window_descriptor.limit is not None
+        and over_window_descriptor.window_size is not None
+    ):
+        # TODO Adds aggregations on both limited and timed window
+        ...
+
+    if agg_func == AggFunc.AVG:
+        result = functions.avg(expr)
+    elif agg_func == AggFunc.MIN:
+        result = functions.min(expr)
+    elif agg_func == AggFunc.MAX:
+        result = functions.max(expr)
+    elif agg_func == AggFunc.SUM:
+        result = functions.sum(expr)
+    elif agg_func == AggFunc.FIRST_VALUE:
+        result = functions.first(expr)
+    elif agg_func == AggFunc.LAST_VALUE:
+        result = functions.last(expr)
+    elif agg_func == AggFunc.ROW_NUMBER:
+        result = functions.row_number()
+    elif agg_func == AggFunc.COUNT:
+        result = functions.count(expr)
+    elif agg_func == AggFunc.VALUE_COUNTS:
+        # TODO Adds VALUE_COUNTS support for SparkProcessor
+        raise FeathubTransformationException(
+            "VALUE_COUNTS is not supported for SparkProcessor currently."
+        )
+    else:
+        raise FeathubTransformationException(
+            f"Unsupported aggregation for SparkProcessor {agg_func}."
+        )
+
+    return result.over(window_spec)
+
+
+def evaluate_over_window_transform(
+    dataframe: DataFrame,
```

Comment:
PR https://github.com/alibaba/feathub/pull/57 aliased DataFrame as NativeSparkDataFrame to avoid confusion with pandas DataFrame. Can we do the same here for consistency?